cmd/runtimetest/main.go

```diff
@@ -191,6 +191,20 @@ func validateSysctls(spec *rspec.Spec) error {
 	return nil
 }
 
+func validateROPaths(spec, *rspec.Spec) error {
+	fmt.Println("validating Readonly Paths")
+	for _, v := range spec.Linux.ReadonlyPaths {
+		fi, err := os.Stat(v)
```

Comment:
On Tue, May 24, 2016 at 09:35:14PM -0700, Ma Shimiao wrote:
> > @@ -191,6 +191,20 @@ func validateSysctls(spec *rspec.Spec) error {
> >  	return nil
> >  }
> >  
> > +func validateROPaths(spec, *rspec.Spec) error {
> > +	fmt.Println("validating Readonly Paths")
> > +	for _, v := range spec.Linux.ReadonlyPaths {
> > +		fi, err := os.Stat(v)
>
> I got from spec `readonlyPaths will set the provided paths as
> readonly inside the container.`

Right, but unless you're joining the container's mount namespace, you
may not be looking inside the container.  Consider the case where the
container creates a new mount namespace, mounts a new /proc, and lists
/proc/cpuinfo (or whatever) in readonlyPaths (which the runtime
decides to implement by bind-mounting with a ro option).  That's going
to still be read/write in the *runtime* mount namespace [1]; it's only
read-only in the *container* mount namespace [2].

[1]: https://github.com/opencontainers/runtime-spec/blob/v0.5.0/glossary.md#runtime-namespace
[2]: https://github.com/opencontainers/runtime-spec/blob/v0.5.0/glossary.md#container-namespace